plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java

```diff
@@ -954,6 +962,24 @@ public ListObjectsV2Result listObjectsV2(ListObjectsV2Request listObjectsV2Reque
         }
     }
 
+    @Test
+    public void testThatTrinoS3FileSystemReportsConsumedMemory()
+            throws IOException
+    {
+        AggregatedMemoryContext memoryContext = mock(AggregatedMemoryContext.class);
```

Comment:
you don't really have to mock these. Use real objects (e.g `io.trino.memory.context.AggregatedMemoryContext#newRootAggregatedMemoryContext` with reservation handler to track max allocation